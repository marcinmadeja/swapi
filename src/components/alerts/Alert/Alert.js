import React from 'react';
import PropTypes from 'prop-types';
import { CircularProgress } from 'material-ui/Progress';
import { ProgressBar, Main } from './Alert.styled';

const AlertStandard = ({
  msg,
  progressBar,
}) => {
  function renderProgressBar() {
    if (!progressBar) return null;
    return (
      <ProgressBar>
        <CircularProgress
          size={25}
          thickness={2}
        />
      </ProgressBar>
    );
  }

  return (
    <Main>
      {renderProgressBar()}
      {msg}
    </Main>
  );
};

AlertStandard.propTypes = {
  progressBar: PropTypes.bool,
  msg: PropTypes.string.isRequired,
};

AlertStandard.defaultProps = {
  progressBar: false,
};

export default AlertStandard;
