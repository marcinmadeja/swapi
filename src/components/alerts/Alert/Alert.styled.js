import styled from 'styled-components';

export const Alert = styled.div`
	display: flex;
	align-items: center;
	min-height: 44px;
	padding: 5px 25px;
	border: 1px solid transparent;

	color: rgba(0, 0, 0, 0.6);
	font-size: 17px;
	font-weight: 400;
	border-radius: 5px;

	box-shadow: 0px 2px 1px -1px rgba(0, 0, 0, 0.2), 0px 1px 1px 0px rgba(0, 0, 0, 0.14), 0px 1px 3px 0px rgba(0, 0, 0, 0.12);
`;

export const Main = styled.div`
	display: flex;
	align-items: center;
`;

export const AlertStandardStyled = Alert.extend`
	background: #F5F5F5;
`;

export const AlertSuccessStyled = Alert.extend`
	background: #388E3C;
	color: #fff;
`;

export const AlertWarningStyled = Alert.extend`
	background: #F9A825;
	color: #fff;
`;

export const AlertDangerStyled = Alert.extend`
	background: #C62828;
	color: #fff;
`;

export const ProgressBar = styled.div`
	margin-right: 15px;
`;
