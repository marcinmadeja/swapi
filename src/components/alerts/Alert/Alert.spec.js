import React from 'react';
import 'jest-styled-components';
import Alert from './Alert';

const defaultProps = {
  msg: '',
};

const setup = props => {
  props = { ...defaultProps, ...props };
  const component = <Alert {...props} />;
  const shallowComponent = shallow(component);

  return {
    component,
    shallowComponent,
  };
};

describe('Alert', () => {
  it('should render alert with message and progress bar', () => {
    const { component } = setup({ msg: 'Test message', progressBar: true });

    const tree = renderer
      .create(component)
      .toJSON();

    expect(tree).toMatchSnapshot(tree);
  });
});
