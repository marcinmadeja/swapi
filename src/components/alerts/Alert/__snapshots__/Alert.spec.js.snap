// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[object Object] 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  margin-right: 15px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="MuiCircularProgress-root-1 MuiCircularProgress-colorPrimary-2"
      role="progressbar"
      style={
        Object {
          "height": 25,
          "width": 25,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg-4 MuiCircularProgress-svgIndeterminate-5"
        viewBox="0 0 50 50"
      >
        <circle
          className="MuiCircularProgress-circle-6 MuiCircularProgress-circleIndeterminate-7"
          cx={25}
          cy={25}
          fill="none"
          r={20}
          strokeWidth={2}
          style={Object {}}
        />
      </svg>
    </div>
  </div>
  Test message
</div>
`;
